src/ZipFile.jl

```diff
@@ -352,7 +358,7 @@ function close(w::Writer)
 	_writele(w._io, @compat UInt32(cdpos))
 	_writele(w._io, @compat UInt16(0))
 
-	if !isa(w._io, IOBuffer)
+	if w._close_io
```

Comment:
I was wrong. Flush usually means to "clear the write buffer" but further writes should still be possible. However, if we write the central directory record when flushing, we won't be able to add more files to the zip file. We should be closing IO only if Reader/Writer owns the IO instance (i.e. we opened the file).

I'll take care of this one myself, so no worries.